libvast/src/system/meta_index.cpp

```diff
@@ -297,18 +297,25 @@ std::vector<uuid> meta_index_state::lookup_impl(const expression& expr) const {
           };
           return search(pred);
         },
-        [&](const type_extractor& lhs, const data&) -> result_type {
+        [&](const type_extractor& lhs, const data& d) -> result_type {
           auto result = [&] {
             if (caf::holds_alternative<none_type>(lhs.type)) {
               VAST_ASSERT(!lhs.type.name().empty());
               auto pred = [&](auto& field) {
-                return field.type.name() == lhs.type.name();
+                const auto* p = &field.type;
+                while (const auto* a = caf::get_if<alias_type>(p)) {
```

Comment:
Exactly how I envisioned it. I wonder whether we should add our usual protection against overflow by recursion.

My suggestion: write a story for that and add a todo where you'd like to perform the recursion check.

Reply:
There is no recursive call in here.

Regarding the issue at large: I don't think adding recursion limits for all functions that operate on tree types like `type`, `data` or `expression` is a good approach. It would be preferable to me if we could keep it to the functions that construct or alter these types.